### PR TITLE
Test CI against PHP 8.1 & fix deprecation

### PR DESCRIPTION
**Pull request by jmsche**

### CODE REVIEW

**Comment by lyrixx** [comment]
This super cool ! 

I'm wondering: Do we really need all the combinaisons in the matrix? ATM, the matrix is quite big and burn some energy for nothing. 

IMHO the following matrix is enough:

PHP 7.4 + SF 4.4 + lower deps
PHP 7.4 + SF 5.4
PHP 8.0 + SF 6
PHP 8.1 + SF 6

What do you think ?

File: src/IndexNameMapper.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function getPureIndexName(string $fullIndexName): string
             return $matches[1];
         }
 
-        $prefixLength = \strlen($this->prefix);
+        $prefixLength = \strlen((string) $this->prefix);
```
Comment (lyrixx) [inline src/IndexNameMapper.php:68]: I think the following code is more semantic and avoid type juggling
```suggestion
        $prefixLength = $this->prefix ? \strlen($this->prefix) : 0;
```

**Comment by jmsche** [comment]
Okay, I'm going to adapt it :)

**Comment by jmsche** [comment]
Ready for review 👍

**Review by lyrixx — APPROVED** [review]
Super 👍🏼

**Comment by lyrixx** [comment]
Thanks

**Pull request merged**